17.0 accounting vendorbills digitization ocr afma (#13928)
odoo/documentation: content/applications/finance/accounting/vendor_bills/invoice_digitization.rst

=====================
Document digitization
=====================

Document digitization refers to the process of converting paper or digital documents into records
in a database. Using :abbr:`OCR (optical character recognition)` and artificial intelligence
technologies, Odoo reads the content and automatically creates and fills in the records. This
process is mainly used for vendor bills (or refunds).

.. note::

>    Although less common, this digitization process can also be applied to customer invoices/credit
>    notes. :ref:`Settings <accounting/bill-digitization/configuration>` need to be adjusted
>    accordingly.

Reviewer: ```suggestion
   Although less common, this digitization process can also be applied to customer invoices and credit
   notes. The :ref:`settings <accounting/bill-digitization/configuration>` need to be adjusted
   accordingly.
```
nitpicking

Depending on the :ref:`settings <accounting/bill-digitization/configuration>`, documents are either
automatically digitized or require manual processing if digitization is set to on-demand only.

To manually digitize an :ref:`uploaded document
<accounting/bill-digitization/vendor-bills-upload>`, click :guilabel:`Digitize document`.

Once the document has been digitized, a blue banner :guilabel:`All fields will be automated by
Artificial Intelligence, it might take 5 seconds` appears; click :icon:`oi-arrow-right`
:guilabel:`Refresh` if needed. Review and correct any information uploaded during digitization:
click on the related field(s) to edit and make updates, or click :guilabel:`Reload AI data` to
refresh the data.

Then, click :guilabel:`Confirm` to post the document.

.. tip::
   Once a document has been digitized, the :guilabel:`Vendor` field remains empty if the vendor is
   not already in the database. To add it, click the :icon:`fa-caret-down` :guilabel:`(down arrow)`
   in the :guilabel:`Vendor` field; the vendor name appears highlighted in the document preview on
   the right. Click it to open a new vendor form with the name pre-filled.

.. note::

>    The following vendor bill fields are read by OCR:

Reviewer: Could we find a better term than *read* here maybe?
```suggestion
   The following vendor bill fields are recognized by OCR:
```

   - :guilabel:`Vendor`, :guilabel:`Bill Reference`, :guilabel:`Bill Date`, :guilabel:`Payment
     Reference` (only in the Belgian +++xxx/xxxx/xxxxx+++ format), :guilabel:`Recipient Bank`,
     :guilabel:`Due Date`, and the currency (in a multi-currency environment and if the currency is

>      activated).

Reviewer: could we add a doc link for multi-currency here?

   - From the :guilabel:`Invoices Lines` tab: :guilabel:`Product` description/label,

>      :guilabel:`Quantity`, unit :guilabel:`Price`, :guilabel:`Taxes` (if the tax is activated -
>      except India localization), :guilabel:`Untaxed Amount`, and :guilabel:`Total`.

Reviewer: ```suggestion
     :guilabel:`Quantity`, unit :guilabel:`Price`, :guilabel:`Taxes` (if taxes are activated; this is not applicable for
    the Indian localization), :guilabel:`Untaxed Amount`, and :guilabel:`Total`.
```
However, I'm not sure what you mean by *activating taxes*?
and I'd add a doc link to the Indian loca doc page and for taxes, maybe?

.. _accounting/bill-digitization/vendor-bills-matching-po:

> Vendor bills matching with purchase orders

Reviewer: ```suggestion
Vendor bill matching with purchase orders
```

------------------------------------------

> When a digitized vendor bill is read by OCR, Odoo first searches into the database for a matching

Reviewer: ```suggestion
When a digitized vendor bill is read by OCR, Odoo first searches the database for a matching
```

> purchase order and its number. If found, it pulls in as much information as possible from the
> purchase order to the vendor bill. For example, the vendor :guilabel:`Bill Reference` number is
> overwritten by the :guilabel:`Vendor Reference` from the purchase order.

Reviewer: ```suggestion
purchase order and its number. If a match is found, the system copies as much information as possible from the
purchase order to the vendor bill. For example, the vendor :guilabel:`Bill Reference` number is
overwritten by the :guilabel:`Vendor Reference` from the purchase order.
```
not sure about *copy* but I wanted to replace *pulls in*, feel free to replace it if you have a better idea

.. note::
   - All changes are logged in the chatter.
   - A smart button linking to the related purchase order is available on the vendor bill.

To correct any mismatches, either edit the incorrect section of the purchase order or leave it

> blank.

Reviewer: Did you mean vendor bill here?
and I don't understand how leaving it blank would help correct mismatches :see_no_evil:, could you please explain this to me?

.. tip::
   - Electronic vendor bills with embedded XML ensure more accurate and efficient processing.
   - Alternatively, the :ref:`Auto-complete <accounting/vendor_bills/bill-completion>` feature
     can be used to transfer information from the purchase order to the vendor bill, without
     requiring OCR.

.. _accounting/bill-digitization/pricing:

Pricing
=======

The document digitization feature is an In-App Purchase (IAP) service requiring prepaid credits.
Digitizing one document uses one credit.

To buy credits, :ref:`go to the Settings app <iap/buying_credits>` or :menuselection:`Accounting -->
Configuration --> Settings`, navigate to the :guilabel:`Digitization` section, and click
:guilabel:`Buy credits`.

.. note::
   - Odoo Enterprise users with a valid subscription get free credits to test IAP features before
     purchasing more credits for the database. This includes demo/training databases, educational
     databases, and one-app-free databases.

>    - XML files don't require OCR credits because they contain structured code Odoo can read
>      directly, without AI processing.

Reviewer: ```suggestion
   - XML files don't require OCR credits because they contain structured data that can be processed
     directly, without AI.
```
or should be say OCR?

either digitized automatically or requires manual processing.

To manually digitize an :ref:`uploaded vendor bill
<accounting/invoice-digitization/vendor-bills-invoices-upload>`, follow these steps:

#. In the uploaded vendor bill, click :guilabel:`Digitize document`.
#. A blue :guilabel:`All fields will be automated by Artificial Intelligence, it might take 5
   seconds` banner appears; click :icon:`oi-arrow-right` :guilabel:`Refresh` if needed.
#. If necessary, review and correct the information uploaded during digitization. Click on the
   related field(s) to edit and make updates, or click :guilabel:`Reload AI data` to refresh the
   data.
#. Click :guilabel:`Confirm` to post the document.

This process helps the AI learn and improves data recognition for future digitizations.

.. _accounting/invoice-digitization/vendor-bills-matching-po:

Vendor bills matching with purchase orders
------------------------------------------

When a purchase order is detected on a vendor bill through OCR, Odoo searches the database for a
matching purchase order number. If found, it pulls in as much information as possible from the

> purchase order and replaces the OCR-detected data on the vendor bill.

Reviewer: I’m having trouble understanding how this works exactly; I wasn’t able to test it based on the current info or the task description :disappointed:. More specifically:
1) When does Odoo start looking for a potential match in the POs? Once the (digitized) bill is confirmed?
2) I see in the Knowledge article linked in the [task](https://www.odoo.com/odoo/project.task/4865418) the PO needs to meet a few condidtions (e.g., the Billing status) for it to be considered a potential match, should we mention those conditions? Are there others?
3) Then what happens? And what happens if there are several potential matches?
4) Just in case because I'm not 100% sure: Is this only for digitized documents or is this matching feature also available when we create vendor bills the "standard" (i.e. 100% manual) way?

I suggest reviewing this section a bit to make sure everything is clear and accurate.

For example, the vendor :guilabel:`Bill Reference` number is overwritten by the :guilabel:`Vendor
Reference` from the purchase order.

.. note::
   - All changes are logged in the chatter.
   - A smart button linking to the related purchase order is available on the vendor bill.

To correct any mismatches, either edit the incorrect section of the purchase order or leave it
blank.

.. tip::
   - Use electronic invoices with embedded XML to ensure more accurate and efficient processing.

>    - Alternatively, the :ref:`Auto-complete <accounting/vendor_bills/bill-completion>` feature
>      transfers information from the purchase order to the vendor bill without OCR required.

Reviewer: ```suggestion
   - Alternatively, the :ref:`Auto-complete <accounting/vendor_bills/bill-completion>` feature
     can be used to transfer information from the purchase order to the vendor bill, without requiring OCR.
```
